src/matcha_ml/templates/base_template.py

```diff
@@ -0,0 +1,202 @@
+
+        Args:
+            template_src (str): Path of the template source directory.
+            extension (str): File extension to filter files.
+            destination (str): Destination path to copy the files to.
+        """
+        files = glob.glob(os.path.join(template_src, extension))
+        self.copy_files(files, destination)
+
+    def build_template(
+        self,
+        config: TemplateVariables,
+        template_src: str,
+        destination: str,
+        verbose: Optional[bool] = False,
+    ) -> None:
+        """Build and copy the template to the project directory.
+
+        Args:
+            config (TemplateVariables): variables to apply to the template.
+            template_src (str): path of the template to use.
+            destination (str): destination path to write template to.
+            verbose (bool, optional): additional output is shown when True. Defaults to False.
+
+        Raises:
+            MatchaPermissionError: when there are no write permissions on the configuration destination
+        """
+        try:
+            print_status(build_status("\nBuilding configuration template..."))
+
+            # Override configuration if it already exists
+            if os.path.exists(destination):
+                rmtree(destination)
+
+            os.makedirs(destination, exist_ok=True)
+
+            if verbose:
+                print_status(
+                    build_substep_success_status(
+                        f"Ensure template destination directory: {destination}"
+                    )
+                )
+
+            self.copy_main_module_files(template_src, destination)
+            self.copy_submodule_files(template_src, destination, verbose)
+            self.copy_files_with_extension(template_src, "*.tf", destination)
+
+            if verbose:
+                print_status(
+                    build_substep_success_status("Configurations were copied.")
+                )
+
+            configuration_destination = os.path.join(
+                destination, "terraform.tfvars.json"
+            )
+            with open(configuration_destination, "w") as f:
+                json.dump(vars(config), f)
+
+            if verbose:
+                print_status(
+                    build_substep_success_status("Template variables were added.")
+                )
+
+        except PermissionError:
+            raise MatchaPermissionError(
+                f"Error - You do not have permission to write the configuration. Check if you have write permissions for '{destination}'."
+            )
+
+        if verbose:
+            print_status(
+                build_substep_success_status("Template configuration has finished!")
+            )
+
+        print_status(
+            build_step_success_status(
+                f"The configuration template was written to {destination}"
+            )
+        )
+
+        print()
```

Comment:
Is this print just used for a newline?

Reply:
Yes, not sure what else I can do modify the `status_message_builders.py`.